### PR TITLE
fix(kumactl) Check version after loading config.

### PR DESCRIPTION
**Pull request by parkanzky**
### Summary

The version comparison in kumactl was not working - it would attempt to connect to kuma-cp to retrieve version prior to loading config and that retrieval would silently fail. This fix moves the API call to happen after config loads.

Also changed the error message to conform to what is specified in the ticket.

### Full changelog

* Fix kumactl / kuma-cp version comparison in kumactl.

### CODE REVIEW

File: app/kumactl/cmd/root.go [hunk 1]
```diff
@@ -57,16 +51,25 @@ func NewRootCmd(root *kumactl_cmd.RootContext) *cobra.Command {
...
-			}
 			if root.IsFirstTimeUsage() {
 				root.Runtime.Config = kumactl_config.DefaultConfiguration()
 				if err := root.SaveConfig(); err != nil {
 					return err
 				}
 			}
-			return root.LoadConfig()
+
+			retval := root.LoadConfig()
```
Comment (jakubdyszkiewicz) [inline app/kumactl/cmd/root.go:61]: LoadConfig returns an error that needs to be handled

File: app/kumactl/cmd/root.go [hunk 2]
```diff
@@ -57,16 +51,25 @@ func NewRootCmd(root *kumactl_cmd.RootContext) *cobra.Command {
...
+			client, err := root.CurrentApiClient()
+			if err != nil {
+				kumactlLog.Error(err, "Unable to get index client")
+			} else {
+				kumaBuildVersion, _ = client.GetVersion()
+			}
+
+			if kumaBuildVersion != nil && (kumaBuildVersion.Version != kuma_version.Build.Version || kumaBuildVersion.Tagline != kuma_version.Product) {
+				cmd.Println("WARNING: You are using kumactl version " + kuma_version.Build.Version + " for " + kuma_version.Product + ", but the server returned version: " + kumaBuildVersion.Tagline + " " + kumaBuildVersion.Version)
```
Comment (jakubdyszkiewicz) [inline app/kumactl/cmd/root.go:70]: I'm wondering how does it work with `kumactl install control-plane | kubectl apply -f -`.
kumactl command produces bunch of YAMLs that are piped to kubectl.
Let's say we've got this incompatibility, wouldn't this break the command?

What if we print this on stderr?

Comment (parkanzky) [reply app/kumactl/cmd/root.go:70]: Good call, I just moved the original error message and didn't check that it was going to stdout. Should definitely go to stderr.

File: app/kumactl/cmd/root.go [hunk 3]
```diff
@@ -57,16 +51,29 @@ func NewRootCmd(root *kumactl_cmd.RootContext) *cobra.Command {
...
 			if root.IsFirstTimeUsage() {
 				root.Runtime.Config = kumactl_config.DefaultConfiguration()
 				if err := root.SaveConfig(); err != nil {
 					return err
 				}
 			}
-			return root.LoadConfig()
+
+			err = root.LoadConfig()
+			if err != nil {
```
Comment (jakubdyszkiewicz) [inline app/kumactl/cmd/root.go:62]: nit: you can shorten this expression with `if err := root.LoadConfig(); err != nil {`

**Review by jakubdyszkiewicz — APPROVED** [review]

File: app/kumactl/cmd/root.go (hunk 3 above)
Comment (parkanzky) [reply app/kumactl/cmd/root.go:62]: Will do.

**Pull request merged**